### PR TITLE
Use url.URL to create the pprof URL

### PR DESCRIPTION
**Pull request by prashantv**
To avoid issues like #20, we can use url.URL to create the final URL.

r @sandlerben

### CODE REVIEW

File: main_test.go [hunk 1]
```diff
@@ -69,7 +69,7 @@ func TestCreateAndRunAppDefaultValues(t *testing.T) {
 	torcher.createAndRunApp()
 }
 
-func TestGoTorchCommand(t *testing.T) {
+func testGoTorchCommand(t *testing.T, url string) {
```
Comment (sandlerben) [inline main_test.go]: How come you made test lowercase?

File: main_test.go [hunk 2]
```diff
@@ -97,6 +97,13 @@ func TestGoTorchCommand(t *testing.T) {
 	mockVisualizer.AssertExpectations(t)
 }
 
+func TestGoTorchCommand(t *testing.T) {
+	testGoTorchCommand(t, "http://localhost")
+
+	// Trailing slash in url should still work.
+	testGoTorchCommand(t, "http://localhost/")
```
Comment (sandlerben) [inline main_test.go]: This is a great idea. Is it typical to write test cases which call other test cases?

**Comment by sandlerben** [comment]
@prashantv The coverage went down just because we added one more 
```Go
if err != nil { 
    log.Fatal(err)
} 
```
It's frustrating that we can't test that log.Fatal happens. Do you think it would be worthwhile to wrap `goTorchCommand` inside a function like `commandAction` so that `goTorchCommand` would return errors and we would `log.Fatal` inside `commandAction`?

File: main_test.go (hunk 2 above)
Comment (prashantv) [inline main_test.go]: Yep, table tests are more common, but parameterized tests are also common, e.g.
https://github.com/golang/go/blob/master/src/unicode/utf8/utf8_test.go#L222

In general, we should avoid code duplication, as long as the test is still readable and obvious.

**Comment by prashantv** [comment]
@sandlerben re: coverage, I think it's worth considering moving away from the cli package. It overcomplicates the design and makes it harder to write code that just returns err, where you only have a log.Fatal once.

cli is great when you have a complex cli with multiple commands that have different flags etc. However, go-torch only has a single command, so I think you could write a vastly simplified CLI using:
https://github.com/jessevdk/go-flags

Your main function would parse the options and pass an options struct, and you would have a function like
```go
func run(opts *Options) error {

}
```

It would be strongly typed to pass arguments in tests, the function is a simple interface that returns errors that can be tested, and I think the overall amount of code would be less.

This is a much bigger change, but it might be worth trying out.

**Comment by sandlerben** [comment]
@prashantv Interesting proposal. I'll look into that when I have some time.

**Pull request merged**